File: ai.js
const OpenAI = require("openai");

// Create an instance of the OpenAI API client
const openai = new OpenAI({
    apiKey: process.env.OPENAI_KEY,
});

const sendMessage = async (messages) => {
    try {
        const completion = await openai.chat.completions.create({
            messages,
            model: "gpt-3.5-turbo",
        });

        return completion.choices[0].message.content;
    } catch (error) {
        console.log(error);
        return "Something went wrong!";
    }
};

module.exports = sendMessage;

File: ai.py
import g4f

from g4f.Provider import (
    ChatgptAi,
    Bing
)
from flask import Flask, request, jsonify

app = Flask(__name__)


@app.route('/', methods=['POST'])
def process():
    try:
        data = request.get_json()
        messages = data["messages"]

        response = g4f.ChatCompletion.create(
            model="gpt-3.5-turbo",
            messages=messages,
            provider=g4f.Provider.Bing,
        )

        json = {}
        json["message"] = response

        print(data)

        return jsonify(json)
    except Exception as e:
        print(e)
        return jsonify({"error": str(e)})


if __name__ == '__main__':
    app.run()
